File: emsbackend/src/main/java/net/emsProject/emsbackend/service/EventService.java
package net.emsProject.emsbackend.service;

import net.emsProject.emsbackend.dto.EventDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface EventService {

    // Create event
    EventDto createEvent(EventDto eventDto);

    // Get each event
    EventDto getEventById(Long eventId);

    // Get all events
    List<EventDto> getAllEvents();

    // Update events
    EventDto updateEvent(Long eventId, EventDto updatedEvent);

    // Delete event
    void deteteEvent(Long eventId);

    // image upload
    String uploadImage(Long eventId, MultipartFile file);


}

File: emsfrontend/src/services/EventService.js
import axios from 'axios'

const REST_API_BASE_URL = 'http://localhost:8080/api/events';

export const listEvents = () => {
    return axios.get(REST_API_BASE_URL);
}

export const createEvent = (event) => {
    return axios.post(REST_API_BASE_URL, event);
}

export const getEvent = (eventId) => {
    return axios.get(REST_API_BASE_URL + '/' + eventId);
}

export const updateEvent = (eventId, event) => {
    return axios.put(REST_API_BASE_URL + '/' + eventId, event);
}

export const deleteEvent = (eventId) => {
    return axios.delete(REST_API_BASE_URL + '/' + eventId);
}